### PR TITLE
Fix build on C89. Add support QNX platform

### PR DESCRIPTION
**Pull request by Arenoros**
Fix: 
#2861 src/win/udp.c: Incompatible with Visual C++ 9/C89 build windows 
#865 porting to qnx

Add support support build for old linux kernel.
Function like (inotify_init1, dup3, pipe2, epoll_create1) was added in Linux 2.6.27; library support was added to glibc in version 2.9.

src/win/udp.c:
STATIC_ASSERT (...) for C89 must be used in top of functions

include/uv/errno.h:
EALREADY, EINVAL and EBUSY is same value on QNX

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -91,6 +95,25 @@ list(APPEND uv_cflags ${lint-no-hides-global-msvc})
 list(APPEND uv_cflags ${lint-no-conditional-assignment-msvc})
 list(APPEND uv_cflags ${lint-no-unsafe-msvc})
 
+check_include_file(sys/eventfd.h UV_HAVE_EVENTFD_H)
+check_include_file(sys/statvfs.h UV_HAVE_STATVFS_H)
+check_include_file(sys/statfs.h UV_HAVE_STATFS_H)
+
+#check_function_exists(eventfd UV_HAVE_EVENTFD)
```
Comment (bzoz) [inline CMakeLists.txt:102]: Why this is commented out?

File: src/win/udp.c [hunk 2]
```diff
@@ -774,8 +775,7 @@ int uv__udp_set_source_membership6(uv_udp_t* handle,
     mreq.gsr_interface = 0;
   }
 
-  STATIC_ASSERT(sizeof(mreq.gsr_group) >= sizeof(*multicast_addr));
-  STATIC_ASSERT(sizeof(mreq.gsr_source) >= sizeof(*source_addr));
+
```
Comment (bzoz) [inline src/win/udp.c:778]: Please remove the extra new line here.

File: src/win/util.c [hunk 3]
```diff
@@ -504,8 +505,8 @@ uint64_t uv__hrtime(unsigned int scale) {
    * performance counter interval, integer math could cause this computation
    * to overflow. Therefore we resort to floating point math.
    */
-  double scaled_freq = (double)hrtime_frequency_ / scale;
-  double result = (double) counter.QuadPart / scaled_freq;
+  scaled_freq = (double)hrtime_frequency_ / scale;
```
Comment (bzoz) [inline src/win/util.c:508]: While you are at it, could you please add space between `(double)` and `hrtime_freqency_`?

**Comment by bnoordhuis** [comment]
Apropos the "support old linux kernel" commit, we've dropped support for pre-2.6.32 kernels a while ago and don't intend to revisit that.

If you have a need to run on older kernels, stick with an older version of libuv. I think v1.23.0 is the last release to support them (down to 2.6.9.)

File: CMakeLists.txt (hunk 1 above)
Comment (Arenoros) [reply CMakeLists.txt:102]: sry, forgot remove this after test new check

File: src/win/util.c (hunk 3 above)
Comment (Arenoros) [reply src/win/util.c:508]: ok

**Comment by Arenoros** [comment]
> Could you please split this into two PRs - one for fixing the Windows build and one for QNX?
> 
> The Windows part LGTM with two style nits.

did it
#2870 QNX  
#2869 Fix build for c89 with Visual C++ 9

**Comment by bnoordhuis** [comment]
I'll close this now that there are other PRs.

**Pull request closed without merging**